src/Expander.hs

```diff
@@ -1074,3 +1075,36 @@ interpretMacroAction (MacroActionIdentEq how v1 v2) = do
 interpretMacroAction (MacroActionLog stx) = do
   liftIO $ prettyPrint stx >> putStrLn ""
   pure $ Right (ValueBool False) -- TODO unit type
+
+type NewScopeCheckTask = (Context, SplitCore)
+
+scopeCheckExpand :: ExpanderLocal -> SplitCore -> Expand ()
+scopeCheckExpand localState splitCore = do
```

Comment:
I'd prefer a name other that `splitCore` for this, because that can be seen from the type signature anyway. This is the expression being checked, so I think that `expr` or `toCheck` would be more informative.